src/main/cpp/utilities/wavewriter.h

```diff
@@ -32,39 +33,63 @@
+         * This can be used to write successive buffers into a single file
+         * Note: totalBufSizeToWrite describes the final size of the written
+         * WAV data after all write iterations have completed.
+         */
+        static std::ofstream createWAVStream( const char* outFile, size_t totalBufSizeToWrite,
+                                              int sampleRate, int channels )
+        {
+            std::ofstream stream( outFile, std::ios::binary );
 
-template <typename SampleType>
-void writeWAVData( const char* outFile, SampleType* buf, size_t bufSize,
-                   int sampleRate, short channels )
-{
-    std::ofstream stream( outFile, std::ios::binary );
-    stream.write( "RIFF", 4 );
-    t_streamwrite<int>( stream, 36 + bufSize );
-    stream.write( "WAVE", 4 );
-    stream.write( "fmt ", 4 );
-    t_streamwrite<int>  ( stream, 16 );
-    t_streamwrite<short>( stream, 1 );                                           // Format (1 = PCM)
-    t_streamwrite<short>( stream, channels );                                    // Channels
-    t_streamwrite<int>  ( stream, sampleRate );                                  // Sample Rate
-    t_streamwrite<int>  ( stream, sampleRate * channels * sizeof( SampleType )); // Byterate
-    t_streamwrite<short>( stream, channels * sizeof( SampleType ));              // Frame size
-    t_streamwrite<short>( stream, 8 * sizeof( SampleType ));                     // Bits per sample
-    stream.write( "data", 4 );
-    stream.write(( const char* )&bufSize, 4 );
-    stream.write(( const char* )buf, bufSize );
-}
+            // write the header data
+            stream.write( "RIFF", 4 );
+            t_streamwrite<int>( stream, 36 + totalBufSizeToWrite );
+            stream.write( "WAVE", 4 );
+            stream.write( "fmt ", 4 );
+            t_streamwrite<int>  ( stream, 16 );
+            t_streamwrite<short>( stream, 1 );                                          // Format (1 = PCM)
+            t_streamwrite<short>( stream, ( short ) channels );                         // Channels
+            t_streamwrite<int>  ( stream, sampleRate );                                 // Sample rate
+            t_streamwrite<int>  ( stream, sampleRate * channels * sizeof( short int )); // Byte rate
+            t_streamwrite<short>( stream, channels * sizeof( short int ));              // Frame size
+            t_streamwrite<short>( stream, 8 * sizeof( short int ));                     // Bits per sample
+            stream.write( "data", 4 );
+            stream.write(( const char* )&totalBufSizeToWrite, 4 );
+
+            return stream;
+        }
+
+        /**
+         * Appends the contents of given buffer to given stream
+         */
+        template <typename SampleType>
+        static void appendBufferToStream( std::ofstream& stream, SampleType* buf, size_t bufSize )
+        {
+            stream.write(( const char* )buf, bufSize );
+        }
```

Comment:
Allow appending buffers to a WAV file incrementally. This gives us the benefit that we can load and unload the snippets one at a time and not use/risk exhausting a massive amount of memory.